pyne/partisn.py

```diff
@@ -830,3 +833,166 @@ def strip_mat_name(mat_name):
+    Carlo ray tracing to determine volume fractions of each source cell within
+    each mesh volume element. The PARTISN SOURCF card is returned, as well as
+    the ray-traced volume fractions (dagmc.discretize_geom output), so that ray
+    tracing does not have to be done twice when using this function with
+    write_partisn_input. The supplied mesh is also tagged with the calculated
+    source distribution using this function.
+
+    Parameters:
+    -----------
+    geom : str
+        The DAGMC geometry (.h5m) file containing the geometry of interest.
+    mesh : PyNE Mesh
+        The superimposed Cartesian mesh that will be used to define the source
+        for PARTISN transport.
+    cells : list of ints
+        The cell numbers of DAGMC geometry cells which have non-zero source
+        intensity.
+    spectra : list of list of floats
+        The normalized energy spectrum for each of the cells. If spectra are
+        not normalized, they will be normalized in this function.
+    intensities : list of floats
+        The volumetric intensity (i.e. s^-1 cm^-3) for each geometry cell. 
+    tag_name : str, optional, default = 'src'
+        The name of the tag for which source data will be tagged on the mesh.
+    num_rays : int, optional, default = 10
+        For discretize_geom. The number of rays to fire in each mesh row for 
+        each direction.
+    grid : boolean, optional, default = False
+        For discretize_geom. If false, rays starting points are chosen randomly 
+        (on the boundary) for each mesh row. If true, a linearly spaced grid of
+        starting points is used, with dimension sqrt(num_rays) x sqrt(num_rays). 
+        In this case, "num_rays" must be a perfect square.
+  
+    Returns:
+    --------
+    output : str
+        PARTISN SOURF card representing the requested source
+    dg : record array
+        The output of dagmc.discretize_geom; stored in a one dimensional array, 
+        each entry containing the following
+        fields:
+        :idx: int 
+            The volume element index.
+        :cell: int
+            The geometry cell number.
+        :vol_frac: float
+            The volume fraction of the cell withing the mesh ve.
+        :rel_error: float
+            The relative error associated with the volume fraction.
+        This array is returned in sorted order with respect to idx and cell, with
+        cell changing fastest.
+    """
+    # discretize_geom inputs
+    if 'tag_name' in kwargs:
+        tag_name = kwargs['tag_name']
+    else:
+        tag_name = "src"
+
+    if 'num_rays' in kwargs:
+        num_rays = kwargs['num_rays']
+    else:
+        num_rays = 10
+
+    if 'grid' in kwargs:
+        grid = kwargs['grid']
+    else:
+        grid = False
+
+    # Check lengths of input
+    if len(cells) != len(spectra)  or len(cells) != len(intensities):
+       raise(ValueError, "Cells, spectra, intensities must be the same length")
```

Comment:
This would cause an error because you are trying to raise a tuple, eg.

```console
scopatz@athenaie ~ $ raise(ValueError , "yo")
Traceback (most recent call last):
  File "/home/scopatz/.local/lib/python3.4/site-packages/xonsh/base_shell.py", line 42, in default
    self.execer.exec(code, mode='single', glbs=self.ctx)  # no locals
  File "/home/scopatz/.local/lib/python3.4/site-packages/xonsh/execer.py", line 127, in exec
    return exec(code, glbs, locs)
  File "<xonsh-code>", line 1, in <module>
TypeError: exceptions must derive from BaseException
```

This should probably be 

```python
raise ValueError("Cells, spectra, intensities must be the same length")
```